### PR TITLE
Add a GitHub Action workflow that fails when PRs are labeled with 'DO…

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Add a GitHub Action workflow that fails when PRs are labeled with `DO NOT MERGE`, as a safety net for when non-draft PRs are approved but labeled as do not merge.

Related to:

- This PR was mistakenly merged: #39010
- We need to now revert that change: #39073

This workflow should help to prevent these sorts of things.

### CODE REVIEW

File: .github/workflows/do-not-merge-label-check.yml
Comment (CarnaViire) [inline .github/workflows/do-not-merge-label-check.yml:14]: I'm not sure it will work like that? I mean, it looks like it will only run once when the label is added, so this will never unblock...

I see that in runtime the check is done a bit differently: https://github.com/dotnet/runtime/blob/main/.github/workflows/check-service-labels.yml

But I'm totally not an expert, so I might be wrong :)

Comment (IEvangelist) [reply .github/workflows/do-not-merge-label-check.yml:14]: This works exactly as we'd like it to. I tested in a GitHub Action workflow sandbox repo that I have. It simply skips (thus implicitly succeeds) when unlabeled.

Comment (IEvangelist) [reply .github/workflows/do-not-merge-label-check.yml:14]: Examples

#### Skipped

image

#### Failed

image

Comment (CarnaViire) [reply .github/workflows/do-not-merge-label-check.yml:14]: just curious, I see in the examples that you touched "do not merge" label in all the tests. what if you add a different label? now I understand the YML file a bit better, so I see that it executes when any of the labels are added or removed, right? So it might "unblock" after you add any other label, despite "do not merge" still being present?

Comment (sharwell) [reply .github/workflows/do-not-merge-label-check.yml:14]: > Skipped

If we can make it ✅ instead of 🚫, that would be nice

**Review by adegeo — APPROVED** [review]

**Pull request merged**
